src/routes/goals/handlers.js

```diff
@@ -227,3 +228,40 @@ export async function retrieveGoalByIdAndRecipient(req, res) {
     await handleErrors(req, res, error, `${logContext}:RETRIEVE_GOAL_BY_ID_AND_RECIPIENT`);
   }
 }
+
+export async function getSimilarGoalsForRecipient(req, res) {
+  const recipientId = parseInt(req.params.recipient_id, 10);
+
+  if (Number.isNaN(recipientId)) {
+    return res.status(400).send('Recipient ID must be an integer');
+  }
+
+  const cluster = Object.prototype.hasOwnProperty.call(req.query, 'cluster');
+  const userId = await currentUserId(req, res);
+  const user = await userById(userId);
+
+  try {
+    const similarGoalIds = await similarGoalsForRecipient(recipientId, cluster);
+
+    const ids = Array.from(similarGoalIds.reduce((acc, resp) => {
+      const id1 = parseInt(resp.goal1.id, 10);
+      const id2 = parseInt(resp.goal2.id, 10);
+      acc.add(id1);
+      acc.add(id2);
+      return acc;
+    }, new Set()));
```

Comment:
won't this produce a one-dimensional array? in other words, we'd never see the case like in the mockup where a recipient has two groups of similar goals, i.e

![Screenshot 2023-11-08 at 8 52 22 AM](https://github.com/HHS/Head-Start-TTADP/assets/3288586/d4ae0bb6-5c69-4679-bfa3-6b36d86f8c4c)

Reply:
ohhhhh ok gotcha, should have read more closely